mitmproxy/addons/serialization/dumpwatcher.py

```diff
@@ -0,0 +1,146 @@
+import sys
+import time
+from functools import wraps
+
+from mitmproxy import ctx
+from mitmproxy import http
+from mitmproxy.io import tnetstring
+from mitmproxy.utils import data
+from mitmproxy.addons.serialization import protobuf
+from mitmproxy.addons.serialization import dummysession
+
+
+def watcher(func):
+    """
+    Decorator for dumpers.
+    Shows how much time it
+    takes to create/retrieve
+    the blob.
+    """
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        start = time.perf_counter()
+        result = func(*args, **kwargs)
+        end = time.perf_counter()
+        ctx.log(f" ===> took {end-start} seconds")
+        return result
+    return wrapper
+
+
+class DumpWatcher:
+
+    """
+        A simple test to measure how different serialization
+        modules behave in dumping and loading flows.
+    """
+
+    default_tpath = data.pkg_data.path('addons/serialization/') + '/dumps-tnet'
+
+    def __init__(self):
+        self.s_modules = ['mitmproxy.io.tnetstring',
+                          'mitmproxy.addons.serialization.protobuf']
+        self.serializers = {}
+        self.deserializers = {}
+        self.rets = {}
+        self.f = None
+        self.session = dummysession.DummySession()
+
+    def _init_dumpers(self):
+        """
+        Apply watcher decorator
+        to dumps/loads methods.
+        """
+        for s in self.s_modules:
+            m = sys.modules[s]
+            if ctx.options.store_dumps:
+                self.serializers[s] = watcher(m.dump)
+                self.deserializers[s] = watcher(m.load)
+            else:
+                self.serializers[s] = watcher(m.dumps)
+                self.deserializers[s] = watcher(m.loads)
+
+    def _run_dumps(self):
+        """
+        Dump flow and collect
+        performance informations.
+        """
+        if self.f:
+            for s in self.serializers:
+                ctx.log('{} module: '.format(s))
```

Comment:
You used f-strings in all other places. Do you want to use f-string here as well? :)

Reply:
Good catch!